### ISSUE TITLE
Microsoft.CodeAnalysis.Formatting.Formatter don't use NewLineForMembersInObjectInit option

### ISSUE THREAD

**Issue by FrediKats** [dotnet/roslyn#39837]
**Version Used**: 
Microsoft.CodeAnalysis.CSharp - 3.3.1
Microsoft.CodeAnalysis.CSharp.Scripting - 3.3.1
Microsoft.CodeAnalysis.CSharp.Workspaces - 3.3.1
Microsoft.CodeAnalysis.Workspaces.Common - 3.3.1

**Steps to Reproduce**:

1. Generate CompilationUnitSyntax with method with Object Initializer Syntax like
```C#
public BusinessEntityAddressEntity Parse(SerializedTransferObject transferObject)
{
            return new BusinessEntityAddressEntity { BusinessEntityId = SPerson.BusinessEntityAddress.BusinessEntityId.GetFrom(transferObject), AddressId = SPerson.BusinessEntityAddress.AddressId.GetFrom(transferObject), AddressTypeId = SPerson.BusinessEntityAddress.AddressTypeId.GetFrom(transferObject), Rowguid = SPerson.BusinessEntityAddress.Rowguid.GetFrom(transferObject), ModifiedDate = SPerson.BusinessEntityAddress.ModifiedDate.GetFrom(transferObject), };
}
```
2. Use Microsoft.CodeAnalysis.AdhocWorkspace, set `CSharpFormattingOptions.NewLineForMembersInObjectInit` as true:

```C#
using (var workspace = new AdhocWorkspace())
 {
     OptionSet options = workspace.Options;
     options = options.WithChangedOption(CSharpFormattingOptions.NewLineForMembersInObjectInit, true);
...
}
```
3. Use Microsoft.CodeAnalysis.Formatting.Formatter with AdhocWorkspace workspace and OptionSet options for p.2:
```C#
SyntaxNode formatted = Formatter.Format(content, workspace, options);
```

**Expected Behavior**:
Formatter will wrap line because of CSharpFormattingOptions.NewLineForMembersInObjectInit:
```C#
public BusinessEntityAddressEntity Parse(SerializedTransferObject transferObject)
{
    return new BusinessEntityAddressEntity
    {
        BusinessEntityId = SPerson.BusinessEntityAddress.BusinessEntityId.GetFrom(transferObject),
        AddressId = SPerson.BusinessEntityAddress.AddressId.GetFrom(transferObject),
        AddressTypeId = SPerson.BusinessEntityAddress.AddressTypeId.GetFrom(transferObject),
        Rowguid = SPerson.BusinessEntityAddress.Rowguid.GetFrom(transferObject),
        ModifiedDate = SPerson.BusinessEntityAddress.ModifiedDate.GetFrom(transferObject),
    };
}
```

**Actual Behavior**:
Object initialize in one line
```C#
public BusinessEntityAddressEntity Parse(SerializedTransferObject transferObject)
{
            return new BusinessEntityAddressEntity { BusinessEntityId = SPerson.BusinessEntityAddress.BusinessEntityId.GetFrom(transferObject), AddressId = SPerson.BusinessEntityAddress.AddressId.GetFrom(transferObject), AddressTypeId = SPerson.BusinessEntityAddress.AddressTypeId.GetFrom(transferObject), Rowguid = SPerson.BusinessEntityAddress.Rowguid.GetFrom(transferObject), ModifiedDate = SPerson.BusinessEntityAddress.ModifiedDate.GetFrom(transferObject), };
}
```

P.S. Full code for step 2 and 3 you can find here:
https://github.com/InRedikaWB/Fluda/blob/master/Fluda.CodeGeneration/CodeGeneration/FilesGenerator.cs#L69-L83

**Comment by InRedikaWB** [comment]
Okay, i found it. And it's look like "by design".

https://github.com/dotnet/roslyn/blob/cbf693f8c486a020ba513e6a1af44bec6e7b0059/src/Workspaces/CSharp/Portable/Formatting/Rules/NewLineUserSettingFormattingRule.cs#L180-L191

Why "we never force it to move up unless it is already on same line"? It's mean i can't force new line for each property in Object Initialization Expression, isn't it? And if it's true - NewLineForMembersInObjectInit is bad name for this option.

**Comment by CyrusNajmabadi** [comment]
> Why "we never force it to move up unless it is already on same line"

Because that's the behavior we found people overwhelmingly wanted.  i.e. when we wrote the formatting engine several times in the past we found that people would want formatting options for things, but would almost always want a special case of "but if its on the same line, don't make it multi-line".  So we codified that directly as a well known behavior 

> NewLineForMembersInObjectInit is bad name for this option.

It's a public API.  Renaming it would break people.
